src/primitives/transaction.h

```diff
@@ -214,8 +213,15 @@ struct CMutableTransaction;
  * - uint32_t nLockTime
  */
 template<typename Stream, typename TxType>
-inline void UnserializeTransaction(TxType& tx, Stream& s) {
-    const bool fAllowWitness = !(s.GetVersion() & SERIALIZE_TRANSACTION_NO_WITNESS);
+inline void UnserializeTransaction(TxType& tx, Stream& s)
+{
+    UnserializeTransaction(tx, s, s.GetParams());
```

Comment:
I think it'd be better to just update the rpc layer if that behaviour's worth keeping.

I had a look at updating `CDataStream` in net but didn't think it was worth including in this draft -- I don't think there's anything subtle/bikesheddy about it that warrants much discussion, and it touches the V2 stuff so seemed like it would just prompt extra rebasing.